tests/devices/unit_tests/test_aperture_scatterguard.py

```diff
@@ -47,3 +178,51 @@ def test_aperture_scatterguard_returns_status_if_within_tolerance(
         0, 0, 1, 0, 0
     )
     assert isinstance(status, StatusBase)
+
+
+def install_logger_for_aperture_and_scatterguard(aperture_scatterguard):
+    parent_mock = MagicMock()
+    mock_ap_x = MagicMock(aperture_scatterguard.aperture.x.set)
+    mock_ap_y = MagicMock(aperture_scatterguard.aperture.y.set)
+    mock_ap_z = MagicMock(aperture_scatterguard.aperture.z.set)
+    mock_sg_x = MagicMock(aperture_scatterguard.scatterguard.x.set)
+    mock_sg_y = MagicMock(aperture_scatterguard.scatterguard.y.set)
+    aperture_scatterguard.aperture.x.set = mock_ap_x
+    aperture_scatterguard.aperture.y.set = mock_ap_y
+    aperture_scatterguard.aperture.z.set = mock_ap_z
+    aperture_scatterguard.scatterguard.x.set = mock_sg_x
+    aperture_scatterguard.scatterguard.y.set = mock_sg_y
+    parent_mock.attach_mock(mock_ap_x, "_mock_ap_x")
+    parent_mock.attach_mock(mock_ap_y, "_mock_ap_y")
+    parent_mock.attach_mock(mock_ap_z, "_mock_ap_z")
+    parent_mock.attach_mock(mock_sg_x, "_mock_sg_x")
+    parent_mock.attach_mock(mock_sg_y, "_mock_sg_y")
+    return parent_mock
+
+
+def compare_actual_and_expected_calls(actual_calls, expected_calls):
+    # ideally, we could use MagicMock.assert_has_calls but a) it doesn't work properly and b) doesn't do what I need
+    i_actual = 0
+    for i_expected in range(0, len(expected_calls)):
```

Comment:
Nit: I think it's a bit cleaner to do:
```python
for expected, actual in zip(expected_calls, actual_calls):
  ...
```
Rather than maintaining the indicies